doc/specs/#6049 - Font config objects.md

```diff
@@ -0,0 +1,98 @@
+
+This design is in line with the way we handle `defaultAppearance` and `unfocusedAppearance`, and was deliberately chosen so we maintain consistency across these different classes.
+
+## UI/UX Design
+
+Users will be able to add font configurations to their profiles that will look something like this:
+
+```
+"font": // this is the default
+{
+    "fontFace": "Cascadia Mono",
+    "fontWeight": "Normal",
+    "fontSize": 12
+},
+"boldFont":
+{
+    "fontFace": "Consolas",
+    "fontWeight": "Extra-Bold"
+}
+"italicFont":
+{
+    "fontWeight": "Light"
+}
+```
+
+In the settings UI, the current way font settings are presented to the user can be preseved, and additional dropdowns can be added for `boldFont` and `italicFont`. Each dropdown will contain options to configure `fontFace` and `fontWeight`.
+
+## Further implementation details
+
+Currently, the `DxRenderer` is given a desired font and it creates a text layout for the default case and a separate text layout for the italic case. To support font configuration objects, the `DxRenderer` will need to be updated to be able to receive a collection of desired fonts, and it will create a separate text layout for each desired font. These text layouts will then be obtained by the renderer when needed, as it already does for the italic case.
+
+## Capabilities
+
+### Accessibility
+
+Should not affect accessibility.
+
+### Security
+
+Should not affect security.
+
+### Reliability
+
+We have had several issues when parsing `fontFace` from settings in the past. This will add further potential for parsing errors simply because there will be more places where we will need to parse `fontFace`. Unfortunate as it is, it is an unavoidable aspect of this feature.
+
+### Compatibility
+
+This feature changes the way we expect to parse font settings. However, we have to make sure we are still able to parse font settings the way they are currently, so as to not break functionality for legacy users.
+
+### Performance, Power, and Efficiency
+
+Rendering several different font faces will probably have a performance impact. Again, this is an unavaoidable aspect of this feature.
+
+## Potential Issues
+
+Already covered in the sections above.
+
+## Future considerations
+
+Similar to the discussion we had regarding appearance configurations, we will need to figure out how we want to deal with overlapping configurations. I.e. if font configurations are defined for both bold and italic, how should font that is both bold and italic be handled?

```

Comment:
_Technically_, text can't be both bold and italic, right? Because terminals would only say if text should be "more bright" or "more faint", I think? But I'm also confused because the API for `TextAttribute` has `IsBold` and `IsFaint` as two separate variables, so theoretically somebody could call `SetIsBold(true)` and `SetIsFaint(true)` back to back, right?

Reply:
Dumb idea, if text is both "more bright" and "more faint", would that mean that it negates itself and you just see text at normal brightness? In other words, would `IsBold() && IsFaint()` mean that we should just use the default font config?

Reply:
I think text can be both bold and italic, try doing:

`echo ^[[1;3m` in command prompt, you get text that's both bold and italic

Reply:
Oof. We almost need people to be able to define a "BoldItalic" variant too.